// expo-health-kit/build/types/index.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.HealthKitError = exports.HealthKitErrorCode = exports.HealthKitDataType = void 0;
var HealthKitDataType;
(function (HealthKitDataType) {
    // Activity and Fitness
    HealthKitDataType["STEPS"] = "HKQuantityTypeIdentifierStepCount";
    HealthKitDataType["DISTANCE_WALKING_RUNNING"] = "HKQuantityTypeIdentifierDistanceWalkingRunning";
    HealthKitDataType["DISTANCE_CYCLING"] = "HKQuantityTypeIdentifierDistanceCycling";
    HealthKitDataType["DISTANCE_SWIMMING"] = "HKQuantityTypeIdentifierDistanceSwimming";
    HealthKitDataType["ACTIVE_ENERGY_BURNED"] = "HKQuantityTypeIdentifierActiveEnergyBurned";
    HealthKitDataType["BASAL_ENERGY_BURNED"] = "HKQuantityTypeIdentifierBasalEnergyBurned";
    HealthKitDataType["FLIGHTS_CLIMBED"] = "HKQuantityTypeIdentifierFlightsClimbed";
    HealthKitDataType["NIKE_FUEL"] = "HKQuantityTypeIdentifierNikeFuel";
    HealthKitDataType["APPLE_EXERCISE_TIME"] = "HKQuantityTypeIdentifierAppleExerciseTime";
    HealthKitDataType["APPLE_STAND_TIME"] = "HKQuantityTypeIdentifierAppleStandTime";
    HealthKitDataType["SWIMMING_STROKE_COUNT"] = "HKQuantityTypeIdentifierSwimmingStrokeCount";
    HealthKitDataType["PUSH_COUNT"] = "HKQuantityTypeIdentifierPushCount";
    HealthKitDataType["WORKOUT_MINUTES"] = "HKQuantityTypeIdentifierWorkoutMinutes";
    // Body Measurements
    HealthKitDataType["HEIGHT"] = "HKQuantityTypeIdentifierHeight";
    HealthKitDataType["BODY_MASS"] = "HKQuantityTypeIdentifierBodyMass";
    HealthKitDataType["BODY_MASS_INDEX"] = "HKQuantityTypeIdentifierBodyMassIndex";
    HealthKitDataType["BODY_FAT_PERCENTAGE"] = "HKQuantityTypeIdentifierBodyFatPercentage";
    HealthKitDataType["LEAN_BODY_MASS"] = "HKQuantityTypeIdentifierLeanBodyMass";
    HealthKitDataType["WAIST_CIRCUMFERENCE"] = "HKQuantityTypeIdentifierWaistCircumference";
    // Heart
    HealthKitDataType["HEART_RATE"] = "HKQuantityTypeIdentifierHeartRate";
    HealthKitDataType["RESTING_HEART_RATE"] = "HKQuantityTypeIdentifierRestingHeartRate";
    HealthKitDataType["WALKING_HEART_RATE"] = "HKQuantityTypeIdentifierWalkingHeartRateAverage";
    HealthKitDataType["HEART_RATE_VARIABILITY_SDNN"] = "HKQuantityTypeIdentifierHeartRateVariabilitySDNN";
    HealthKitDataType["HEART_RATE_RECOVERY_ONE_MINUTE"] = "HKQuantityTypeIdentifierHeartRateRecoveryOneMinute";
    HealthKitDataType["ATRIALFIBRILLATION_BURDEN"] = "HKQuantityTypeIdentifierAtrialFibrillationBurden";
    HealthKitDataType["PERIPHERAL_PERFUSION_INDEX"] = "HKQuantityTypeIdentifierPeripheralPerfusionIndex";
    HealthKitDataType["VO2_MAX"] = "HKQuantityTypeIdentifierVO2Max";
    HealthKitDataType["WALKING_SPEED"] = "HKQuantityTypeIdentifierWalkingSpeed";
    HealthKitDataType["WALKING_STEP_LENGTH"] = "HKQuantityTypeIdentifierWalkingStepLength";
    HealthKitDataType["WALKING_ASYMMETRY_PERCENTAGE"] = "HKQuantityTypeIdentifierWalkingAsymmetryPercentage";
    HealthKitDataType["WALKING_DOUBLE_SUPPORT_PERCENTAGE"] = "HKQuantityTypeIdentifierWalkingDoubleSupportPercentage";
    HealthKitDataType["STAIR_ASCENT_SPEED"] = "HKQuantityTypeIdentifierStairAscentSpeed";
    HealthKitDataType["STAIR_DESCENT_SPEED"] = "HKQuantityTypeIdentifierStairDescentSpeed";
    // Vitals
    HealthKitDataType["BLOOD_PRESSURE_SYSTOLIC"] = "HKQuantityTypeIdentifierBloodPressureSystolic";
    HealthKitDataType["BLOOD_PRESSURE_DIASTOLIC"] = "HKQuantityTypeIdentifierBloodPressureDiastolic";
    HealthKitDataType["RESPIRATORY_RATE"] = "HKQuantityTypeIdentifierRespiratoryRate";
    HealthKitDataType["BODY_TEMPERATURE"] = "HKQuantityTypeIdentifierBodyTemperature";
    HealthKitDataType["BASAL_BODY_TEMPERATURE"] = "HKQuantityTypeIdentifierBasalBodyTemperature";
    HealthKitDataType["BLOOD_GLUCOSE"] = "HKQuantityTypeIdentifierBloodGlucose";
    HealthKitDataType["OXYGEN_SATURATION"] = "HKQuantityTypeIdentifierOxygenSaturation";
    HealthKitDataType["BLOOD_ALCOHOL_CONTENT"] = "HKQuantityTypeIdentifierBloodAlcoholContent";
    HealthKitDataType["FORCED_VITAL_CAPACITY"] = "HKQuantityTypeIdentifierForcedVitalCapacity";
    HealthKitDataType["FORCED_EXPIRATORY_VOLUME1"] = "HKQuantityTypeIdentifierForcedExpiratoryVolume1";
    HealthKitDataType["PEAK_EXPIRATORY_FLOW_RATE"] = "HKQuantityTypeIdentifierPeakExpiratoryFlowRate";
    HealthKitDataType["ENVIRONMENTAL_AUDIO_EXPOSURE"] = "HKQuantityTypeIdentifierEnvironmentalAudioExposure";
    HealthKitDataType["HEADPHONE_AUDIO_EXPOSURE"] = "HKQuantityTypeIdentifierHeadphoneAudioExposure";
    // Nutrition
    HealthKitDataType["DIETARY_ENERGY_CONSUMED"] = "HKQuantityTypeIdentifierDietaryEnergyConsumed";
    HealthKitDataType["DIETARY_CARBOHYDRATES"] = "HKQuantityTypeIdentifierDietaryCarbohydrates";
    HealthKitDataType["DIETARY_FIBER"] = "HKQuantityTypeIdentifierDietaryFiber";
    HealthKitDataType["DIETARY_SUGAR"] = "HKQuantityTypeIdentifierDietarySugar";
    HealthKitDataType["DIETARY_FAT_TOTAL"] = "HKQuantityTypeIdentifierDietaryFatTotal";
    HealthKitDataType["DIETARY_FAT_SATURATED"] = "HKQuantityTypeIdentifierDietaryFatSaturated";
    HealthKitDataType["DIETARY_PROTEIN"] = "HKQuantityTypeIdentifierDietaryProtein";
    HealthKitDataType["DIETARY_CHOLESTEROL"] = "HKQuantityTypeIdentifierDietaryCholesterol";
    HealthKitDataType["DIETARY_SODIUM"] = "HKQuantityTypeIdentifierDietarySodium";
    HealthKitDataType["DIETARY_POTASSIUM"] = "HKQuantityTypeIdentifierDietaryPotassium";
    HealthKitDataType["DIETARY_CALCIUM"] = "HKQuantityTypeIdentifierDietaryCalcium";
    HealthKitDataType["DIETARY_VITAMIN_C"] = "HKQuantityTypeIdentifierDietaryVitaminC";
    HealthKitDataType["DIETARY_VITAMIN_D"] = "HKQuantityTypeIdentifierDietaryVitaminD";
    HealthKitDataType["DIETARY_IRON"] = "HKQuantityTypeIdentifierDietaryIron";
    HealthKitDataType["DIETARY_WATER"] = "HKQuantityTypeIdentifierDietaryWater";
    HealthKitDataType["DIETARY_CAFFEINE"] = "HKQuantityTypeIdentifierDietaryCaffeine";
    // Sleep
    HealthKitDataType["SLEEP_ANALYSIS"] = "HKCategoryTypeIdentifierSleepAnalysis";
    HealthKitDataType["SLEEP_DURATION"] = "HKQuantityTypeIdentifierSleepDurationInBed";
    HealthKitDataType["SLEEP_CORE"] = "HKQuantityTypeIdentifierSleepCore";
    HealthKitDataType["SLEEP_DEEP"] = "HKQuantityTypeIdentifierSleepDeep";
    HealthKitDataType["SLEEP_REM"] = "HKQuantityTypeIdentifierSleepREM";
    HealthKitDataType["SLEEP_AWAKE"] = "HKQuantityTypeIdentifierSleepAwake";
    // Mindfulness and Symptoms
    HealthKitDataType["MINDFUL_SESSION"] = "HKCategoryTypeIdentifierMindfulSession";
    HealthKitDataType["MINDFUL_MINUTES"] = "HKQuantityTypeIdentifierMindfulMinutes";
    HealthKitDataType["SYMPTOMS"] = "HKCategoryTypeIdentifierSymptoms";
    HealthKitDataType["MOOD"] = "HKCategoryTypeIdentifierMood";
    // Workouts
    HealthKitDataType["WORKOUT"] = "HKWorkoutTypeIdentifier";
    HealthKitDataType["WORKOUT_ROUTE"] = "HKWorkoutRouteTypeIdentifier";
    // Reproductive Health
    HealthKitDataType["MENSTRUATION"] = "HKCategoryTypeIdentifierMenstrualFlow";
    HealthKitDataType["SPOTTING"] = "HKCategoryTypeIdentifierIntermenstrualBleeding";
    HealthKitDataType["CERVICAL_MUCUS_QUALITY"] = "HKCategoryTypeIdentifierCervicalMucusQuality";
    HealthKitDataType["OVULATION_TEST_RESULT"] = "HKCategoryTypeIdentifierOvulationTestResult";
    HealthKitDataType["PREGNANCY_TEST"] = "HKCategoryTypeIdentifierPregnancyTestResult";
    HealthKitDataType["PROGESTERONE_TEST"] = "HKQuantityTypeIdentifierProgesteroneTestResult";
    HealthKitDataType["SEXUAL_ACTIVITY"] = "HKCategoryTypeIdentifierSexualActivity";
    // Other
    HealthKitDataType["ELECTROCARDIOGRAM"] = "HKDataTypeIdentifierElectrocardiogram";
    HealthKitDataType["HANDWASHING"] = "HKCategoryTypeIdentifierHandwashingEvent";
    HealthKitDataType["TOOTH_BRUSHING"] = "HKCategoryTypeIdentifierToothbrushingEvent";
    HealthKitDataType["FALLS"] = "HKCategoryTypeIdentifierAppleWalkingSteadinessEvent";
    HealthKitDataType["GAIT_STEADINESS"] = "HKQuantityTypeIdentifierAppleWalkingSteadiness";
    HealthKitDataType["INHALER_USAGE"] = "HKCategoryTypeIdentifierInhalerUsage";
    HealthKitDataType["INSULIN_DELIVERY"] = "HKQuantityTypeIdentifierInsulinDelivery";
    HealthKitDataType["TIME_IN_DAYLIGHT"] = "HKQuantityTypeIdentifierTimeInDaylight";
    HealthKitDataType["UV_EXPOSURE"] = "HKQuantityTypeIdentifierUVExposure";
    HealthKitDataType["WATER_TEMPERATURE"] = "HKQuantityTypeIdentifierWaterTemperature";
})(HealthKitDataType || (exports.HealthKitDataType = HealthKitDataType = {}));
var HealthKitErrorCode;
(function (HealthKitErrorCode) {
    HealthKitErrorCode["UNAUTHORIZED"] = "UNAUTHORIZED";
    HealthKitErrorCode["UNAVAILABLE"] = "UNAVAILABLE";
    HealthKitErrorCode["INVALID_CONFIG"] = "INVALID_CONFIG";
    HealthKitErrorCode["EXPORT_FAILED"] = "EXPORT_FAILED";
    HealthKitErrorCode["PERMISSION_DENIED"] = "PERMISSION_DENIED";
    HealthKitErrorCode["INVALID_PARAMETERS"] = "INVALID_PARAMETERS";
    HealthKitErrorCode["BACKGROUND_DELIVERY_FAILED"] = "BACKGROUND_DELIVERY_FAILED";
    HealthKitErrorCode["QUERY_FAILED"] = "QUERY_FAILED";
})(HealthKitErrorCode || (exports.HealthKitErrorCode = HealthKitErrorCode = {}));
class HealthKitError extends Error {
    constructor(code, message, details) {
        super(message);
        this.code = code;
        this.details = details;
        this.name = 'HealthKitError';
        Object.setPrototypeOf(this, HealthKitError.prototype);
    }
}
exports.HealthKitError = HealthKitError;
